spec/helpers/appstore_project_helper.rb

```diff
@@ -0,0 +1,214 @@
+        unless new_blueprint.valid?
+          pp new_blueprint.validate
+          raise 'Cannot create computed attribute'
+        end
+        project.update_from_blueprint(new_blueprint)
+
+        project.attributes(COMPUTED_ATTRIBUTE_ID).drill_down(project.attributes('attr.csv_policies.customer'))
+      end
+
+      def create_reports
+        GoodData.with_project @project.pid do |p|
+          reports = JSON.parse(File.read(REPORTS_FILE)).to_hash['reports']
+          reports.each do |r|
+            report = p.add_report(
+              title: r['title'],
+              left: r['left'].map { |o| GoodData::MdObject[o] },
+              top: r['top'].map { |o| GoodData::MdObject[o] }
+            )
+            report_name = r['title'].to_s.downcase.gsub(/\s/, '.')
+            report.identifier = "report.#{report_name}"
+            report.save
+          end
+
+          ca_report_title = 'Report contains CA'
+          report = p.add_report(title: ca_report_title, left: GoodData::MdObject[COMPUTED_ATTRIBUTE_ID])
+          report_name = ca_report_title.downcase.gsub(/\s/, '.')
+          report.identifier = "report.#{report_name}"
+          report.save
+        end
+      end
+
+      def create_dashboards
+        GoodData.with_project @project.pid do
+          dashboards = JSON.parse(File.read(DASHBOARDS_FILE)).to_hash['dashboards']
+          dashboards.each do |d|
+            dashboard = GoodData::Dashboard.create(
+              { title: d['title'] },
+              client: client, project: @project
+            )
+            d['tabs'].each do |t|
+              tab = dashboard.add_tab(title: t['title'])
+              t['items'].each do |i|
+                tab.add_report_item(report: GoodData::Report[i['identifier']],
+                                    position_x: i['position']['x'],
+                                    position_y: i['position']['y'],
+                                    size_x: i['size']['width'],
+                                    size_y: i['size']['height'])
+              end
+
+              tab.add_report_item(report: GoodData::Report['report.report.contains.ca'],
+                                  position_x: 0,
+                                  position_y: 0,
+                                  size_x: 640,
+                                  size_y:100)
+            end
+            dashboard_name = d['title'].to_s.downcase.gsub(/\s/, '.')
+            dashboard.identifier = "dashboard.#{dashboard_name}"
+            dashboard.add_tag('dashboard')
+            dashboard.save
+          end
+        end
+      end
+
+      def create_tag_for_fact_n_dataset
+        dataset = @project.datasets(DATASET_IDENTIFIER)
+        dataset.add_tag('dataset')
+        dataset.save
+
+        fact = @project.facts(FACT_IDENTIFIER)
+        fact.add_tag('fact')
+        fact.save
+      end
+
+      def ensure_user(login, domain)
+        user = domain.users(login)
```

Comment:
Refers to 'domain' more than self (maybe move it to another class?) (FeatureEnvy)